src/ert/ensemble_evaluator/snapshot.py

```diff
@@ -104,110 +102,181 @@ def convert_iso8601_to_datetime(
-        if self._snapshot is None:
-            raise UnsupportedOperationException(
-                f"updating metadata on {self.__class__} without providing a snapshot"
-                + " is not supported"
-            )
-        dictionary = pyrsistent.pmap({ids.METADATA: metadata})
-        self._data = _recursive_update(self._data, dictionary, check_key=False)
-        self._snapshot.merge_metadata(metadata)
+        self._job_states = defaultdict(dict)
+        """A shallow dictionary of job states. The key is a tuple of three
+        strings with realization id, step id and job id, pointing to a dict with
+        the same members as the Job."""
 
-    def update_real(
-        self,
-        real_id: str,
-        real: "RealizationSnapshot",
-    ) -> None:
-        self._apply_update(SnapshotDict(reals={real_id: real}))
+        self._ensemble_state: Optional[str] = None
+        self._metadata = defaultdict(dict)
 
-    def _apply_update(self, update: "SnapshotDict") -> None:
-        if self._snapshot is None:
-            raise UnsupportedOperationException(
-                f"trying to mutate {self.__class__} without providing a snapshot is "
-                + "not supported"
-            )
-        dictionary = update.dict(
-            exclude_unset=True, exclude_none=True, exclude_defaults=True
+        self._snapshot = snapshot
+
+    @property
+    def status(self) -> str:
+        return self._ensemble_state
+
+    def update_status(self, status: str) -> None:
+        self._ensemble_state = status
+        self._snapshot._my_partial._ensemble_state = status
+
+    def update_metadata(self, metadata: Dict[str, Any]) -> None:
+        self._metadata.update(_filter_nones(metadata))
+        self._snapshot._my_partial._metadata.update(_filter_nones(metadata))
+
+    def update_real(self, real_id: str, real: "RealizationSnapshot") -> None:
+        real_update = _filter_nones(
+            {
+                "status": real.status,
+                "active": real.active,
+                "start_time": real.start_time,
+                "end_time": real.end_time,
+            }
         )
-        self._data = _recursive_update(self._data, dictionary, check_key=False)
-        self._snapshot.merge(dictionary)
+        if real_id not in self._realization_states:
+            self._realization_states[real_id] = {}
+        self._realization_states[real_id].update(real_update)
+        self._snapshot._my_partial._realization_states[real_id].update(real_update)
 
     def update_step(
         self, real_id: str, step_id: str, step: "Step"
     ) -> "PartialSnapshot":
-        if self._snapshot is None:
-            raise UnsupportedOperationException(
-                f"cannot update step for {self.__class__} without providing a snapshot"
-            )
-        self._apply_update(
-            SnapshotDict(reals={real_id: RealizationSnapshot(steps={step_id: step})})
+        step_idx = (real_id, step_id)
```

Comment:
Referring to my previews comment on necessity of steps :)

Reply:
yeah would be awesome to remove steps. i totally wanna do it, too.